### PR TITLE
MM-16438: fixed and slightly improved `/jira transition`

### PR DESCRIPTION
**Pull request by levb**
ticket: https://mattermost.atlassian.net/browse/MM-16438

- Fixed parameter validation
- Improved transition disambiguation
- Also fixed a bug in a prior commit for unmarshalling ASC

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -529,80 +530,78 @@ func (p *Plugin) responsef(commandArgs *model.CommandArgs, format string, args .
...
-// 		return p.responsef(header, "failed to delete Jira instance %s: %v", instanceKey, err)
-// 	}
-
-// 	// if that was our only instance, just respond with an empty list.
-// 	if len(known) == 1 {
-// 		return executeList(p, c, header)
-// 	}
-// 	return executeInstanceSelect(p, c, header, "1")
-// }
+func executeInstanceSelect(p *Plugin, c *plugin.Context, header *model.CommandArgs, args ...string) *model.CommandResponse {
```
Comment (cpoile) [inline server/command.go]: Are we exposing these now? (same for instance delete below)

**Comment by levb** [comment]
I think this got merged before @aaronrothschild review, can you please review it post-factum?

**Comment by aaronrothschild** [comment]
Seems to make sense.  Peronsally, I would expect both "inprogress" and "progress" to work to transition an issue to "in progress" status

File: server/command.go (hunk 1 above)
Comment (levb) [reply server/command.go]: They are still commented out in the table

**Pull request merged**